BiasAndFairnessServers/src/controllers/bias_and_fairness.py

```
@@ -520,14 +520,29 @@ async def run_bias_fairness_evaluation(job_id: int, config_path: str, eval_id: s
+            try:
+                stdout_bytes, stderr_bytes = await _asyncio.wait_for(proc.communicate(), timeout=1800)
+            except _asyncio.TimeoutError:
+                proc.kill()
+                await proc.wait()
+                stdout_bytes, stderr_bytes = b"", b"Timeout"
+            result = type("Obj", (), {
+                "returncode": proc.returncode,
+                "stdout": stdout_bytes.decode(errors="ignore"),
+                "stderr": stderr_bytes.decode(errors="ignore"),
+            })
+        except Exception as e:
```

Ruff (0.14.4)

Do not catch blind exception: `Exception`

(BLE001)